apps/gdalalg_raster_index.cpp

```diff
@@ -0,0 +1,222 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster index" subcommand
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_index.h"
+
+#include "cpl_conv.h"
+#include "gdal_priv.h"
+#include "gdal_utils_priv.h"
+#include "ogrsf_frmts.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALRasterIndexAlgorithm::GDALRasterIndexAlgorithm()        */
+/************************************************************************/
+
+GDALRasterIndexAlgorithm::GDALRasterIndexAlgorithm()
+    : GDALVectorOutputAbstractAlgorithm(NAME, DESCRIPTION, HELP_URL)
+{
+    AddProgressArg();
+    AddInputDatasetArg(&m_inputDatasets, GDAL_OF_RASTER)
+        .SetAutoOpenDataset(false);
+    GDALVectorOutputAbstractAlgorithm::AddAllOutputArgs();
+
+    AddCommonOptions();
+
+    AddArg("source-crs-field-name", 0,
+           _("Name of the field to store the SRS of each dataset"),
+           &m_sourceCrsName)
+        .SetMinCharCount(1);
+    AddArg("source-crs-format", 0,
+           _("Format in which the SRS of each dataset must be written"),
+           &m_sourceCrsFormat)
+        .SetDefault(m_sourceCrsFormat)
+        .SetChoices("auto", "WKT", "EPSG", "PROJ");
+}
+
+/************************************************************************/
+/*          GDALRasterIndexAlgorithm::GDALRasterIndexAlgorithm()        */
+/************************************************************************/
+
+GDALRasterIndexAlgorithm::GDALRasterIndexAlgorithm(
+    const std::string &name, const std::string &description,
+    const std::string &helpURL)
+    : GDALVectorOutputAbstractAlgorithm(name, description, helpURL)
+{
+}
+
+/************************************************************************/
+/*              GDALRasterIndexAlgorithm::AddCommonOptions()            */
+/************************************************************************/
+
+void GDALRasterIndexAlgorithm::AddCommonOptions()
+{
+
+    AddArg("recursive", 0,
+           _("Whether input directories should be explored recursively."),
+           &m_recursive);
+    AddArg("filename-filter", 0,
+           _("Pattern that the filenames in input directories should follow "
+             "('*' and '?' wildcard)"),
+           &m_filenameFilter);
+    AddArg("min-pixel-size", 0,
+           _("Minimum pixel size in term of geospatial extent per pixel "
+             "(resolution) that a raster should have to be selected."),
```

Comment:
Someone better in English than me should check, but the description feels somehow odd to me. BTW if x and y resolutions are not the same, which resolution rules?

Reply:
> BTW if x and y resolutions are not the same, which resolution rules?

a geometric mean of it